amd_openvx_extensions/amd_rpp/source/Downmix.cpp

```diff
@@ -0,0 +1,271 @@
+        // rpp_status = rppt_Downmix_gpu((void *)data->cl_pSrc, data->src_desc_ptr, (void *)data->cl_pDst, data->src_desc_ptr,  data->alpha, data->beta, data->roi_tensor_Ptr, data->roiType, data->handle->rppHandle);
+        return_status = (rpp_status == RPP_SUCCESS) ? VX_SUCCESS : VX_FAILURE;
+#elif ENABLE_HIP
+        refreshDownmix(node, parameters, num, data);
+        // rpp_status = rppt_Downmix_gpu((void *)data->hip_pSrc, data->src_desc_ptr, (void *)data->hip_pDst, data->src_desc_ptr,  data->alpha, data->beta, data->hip_roi_tensor_Ptr, data->roiType, data->handle->rppHandle);
+        return_status = (rpp_status == RPP_SUCCESS) ? VX_SUCCESS : VX_FAILURE;
+#endif
+    }
+    if (data->device_type == AGO_TARGET_AFFINITY_CPU)
+    {
+        refreshDownmix(node, parameters, num, data);
+        rpp_status = rppt_down_mixing_host((float *)data->pSrc, data->src_desc_ptr, (float *)data->pDst, data->dst_desc_ptr, data->samples, data->channels, false);
+        return_status = (rpp_status == RPP_SUCCESS) ? VX_SUCCESS : VX_FAILURE;
+    }
+    return return_status;
+}
+
+static vx_status VX_CALLBACK initializeDownmix(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+    DownmixLocalData *data = new DownmixLocalData;
+    memset(data, 0, sizeof(*data));
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[5], &data->device_type, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxReadScalarValue((vx_scalar)parameters[4], &data->nbatchSize));
+
+    // Querying for input tensor
+    data->src_desc_ptr = &data->srcDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &data->src_desc_ptr->numDims, sizeof(data->src_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, &data->in_tensor_dims, sizeof(vx_size) * data->src_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &data->in_tensor_type, sizeof(data->in_tensor_type)));
+    if (data->in_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->src_desc_ptr->dataType = RpptDataType::F32;
+    data->src_desc_ptr->offsetInBytes = 0;
+
+    // Querying for output tensor
+    data->dst_desc_ptr = &data->dstDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_NUMBER_OF_DIMS, &data->dst_desc_ptr->numDims, sizeof(data->dst_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DIMS, &data->out_tensor_dims, sizeof(vx_size) * data->dst_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1],VX_TENSOR_DATA_TYPE, &data->out_tensor_type, sizeof(data->out_tensor_type)));
+    if (data->out_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+        data->dst_desc_ptr->dataType = RpptDataType::F32;
+    data->dst_desc_ptr->offsetInBytes = 0;
+
+    // source_description_ptr
+    data->src_desc_ptr->n = data->in_tensor_dims[0];
```

Comment:
We are using only first 3 values of in_tensor_dims / out_tensor_dims in this file
in_tensor_dims[0];
in_tensor_dims[1];
in_tensor_dims[2];

We are creating a tensor with 5 Values

#define NUM_OF_DIMS 5
size_t in_tensor_dims[NUM_OF_DIMS];

Are the 5 values need or only 3 is needed? Please check and update accordingly

Reply:
Will check on this

Reply:
Changed to 3